src/graph/graph.c

```diff
@@ -680,10 +674,288 @@ void Graph_DeleteNode(Graph *g, Node *n) {
+            deletion.col = src_id;
+            deletion.row = dest_id;
+            deletions = array_append(deletions, deletion);
+
+            deletion.M = M;
+            deletions = array_append(deletions, deletion);
+        } else {
+            /* Multiple edges connecting src to dest
+             * locate specific edge and remove it
+             * revert back from array representation to edge ID
+             * incase we're left with a single edge connecting src to dest. */
+
+            int i = 0;
+            EdgeID id = ENTITY_GET_ID(e);
+            EdgeID *multi_edges = (EdgeID*)edge_id;
+            int multi_edge_count = array_len(multi_edges);
+
+            // Locate edge within edge array.
+            for(; i < multi_edge_count; i++) {
+                if(multi_edges[i] == id) break;
+            }
+            assert(i < multi_edge_count);
+
+            /* Remove edge from edge array
+             * migrate last edge ID and reduce array size.
+             * TODO: reallocate array of size / capacity ratio is high. */
+            multi_edges[i] = multi_edges[multi_edge_count-1];
+            array_pop(multi_edges);
+
+            /* Incase we're left with a single edge connecting src to dest
+             * revert back from array to scalar. */
+            if(array_len(multi_edges) == 1) {
+                edge_id = multi_edges[0];
+                array_free(multi_edges);
+                GrB_Matrix_setElement_UINT64(M, SET_MSB(edge_id), dest_id, src_id);
+            }
+        }
+
+        // Free and remove edges from datablock.
+        DataBlock_DeleteItem(g->edges, ENTITY_GET_ID(e));
+    }
+
+    // Delete entries.
+    for(int i = 0; i < array_len(deletions); i++) {
+        deletion = deletions[i];
+        assert(GxB_Matrix_Delete(deletion.M, deletion.row, deletion.col) == GrB_SUCCESS);
+    }
+
+    int relationCount = Graph_RelationTypeCount(g);
+    uint deletion_count = array_len(deletions);
+    for(uint i = 0; i < deletion_count; i++) {
+        deletion = deletions[i];
+        GrB_Index src = deletion.col;
+        GrB_Index dest = deletion.row;
+
+        // See if source is connected to destination with additional edges.
+        bool connected = false;
+        for(int i = 0; i < relationCount; i++) {
+            R = Graph_GetRelationMatrix(g, i);
+            info = GrB_Matrix_extractElement_BOOL(&connected, R, dest, src);
+            if(info == GrB_SUCCESS) break;
+        }
+
+        /* There are no additional edges connecting source to destination
+         * Remove edge from THE adjacency matrix. 
+         * It is OK to remove entries from the adjacency matrix, as we're 
+         * not trying to extract entries from it. */
+        if(!connected) {
+            M = Graph_GetAdjacencyMatrix(g);
+            assert(GxB_Matrix_Delete(M, dest, src) == GrB_SUCCESS);
+
+            M = _Graph_Get_Transposed_AdjacencyMatrix(g);
+            assert(GxB_Matrix_Delete(M, src, dest) == GrB_SUCCESS);
```

Comment:
Shouldn't we be updating the relations_map as well?

Reply:
No, it has already been updated,
we're removing the last edge connecting a to b, as a result a is no longer connected to b, all that's left is updating the adjacency matrix